protostar/commands/test/test_result_formatter.py

```diff
@@ -0,0 +1,230 @@
+from pathlib import Path
+from typing import Callable, Dict, List
+
+from protostar.commands.test.test_environment_exceptions import ExceptionMetadata
+from protostar.commands.test.test_output_recorder import OutputName, format_output_name
+from protostar.protostar_exception import (
+    UNEXPECTED_PROTOSTAR_ERROR_MSG,
+    ProtostarException,
+)
+from protostar.utils.log_color_provider import log_color_provider
+
+from .test_results import (
+    BrokenTestSuiteResult,
+    FailedFuzzTestCaseResult,
+    FailedTestCaseResult,
+    PassedFuzzTestCaseResult,
+    PassedTestCaseResult,
+    TestResult,
+    UnexpectedBrokenTestSuiteResult,
+)
+
+LogCallback = Callable[[str], None]
+
+
+# pylint: disable=too-many-return-statements
+def format_test_result(test_result: TestResult) -> str:
+    if isinstance(test_result, PassedFuzzTestCaseResult):
+        return _format_passed_fuzz_test_case_result(test_result)
+    if isinstance(test_result, FailedFuzzTestCaseResult):
+        return _format_failed_fuzz_test_case_result(test_result)
+    if isinstance(test_result, PassedTestCaseResult):
+        return _format_passed_test_case_result(test_result)
+    if isinstance(test_result, FailedTestCaseResult):
+        return _format_failed_test_case_result(test_result)
+    if isinstance(test_result, UnexpectedBrokenTestSuiteResult):
+        return _format_unexpected_exception_test_suite_result(test_result)
+    if isinstance(test_result, BrokenTestSuiteResult):
+        return _format_broken_test_suite_result(test_result)
+    raise ProtostarException("Unexpected test result: ", str(test_result))
```

Comment:
```suggestion
    raise NotImplementedError("Unreachable")
```

this is not protostar error; this is a critical programmer's error and this exception is popular convention from what I have observed